Objects/typeobject.c

```diff
@@ -9565,6 +9565,9 @@ add_tp_new_wrapper(PyTypeObject *type)
         return -1;
     }
     r = PyDict_SetItem(dict, &_Py_ID(__new__), func);
+    if (!r) {
+        _PyObject_SetDeferredRefcount(func);
```

Comment:
Can we move this up before the `PyDict_SetItem`? I'd rather set the deferred reference counting bit before exposing the function to avoid any race conditions on `ob_gc_bits`.